docs/devel/README.md

```diff
@@ -0,0 +1,138 @@
+# Developer Setup
+
+This documents show how to build and run poseidon and other compenets on a dev setup.
+
+* Dependency 
+   * Kubernetes :- Running instance of a [kubernetes cluster](https://kubernetes.io/docs/setup/) is required. 
+   * Firmament  :- For firmament build info please refer [here](https://github.com/camsas/firmament/blob/master/README.md#building-instructions).
+   * Heapster   :- For deploying heapster with poseidon sink. please use these [instruction](https://github.com/shivramsrivastava/poseidon/tree/script_changes/docs/install#steps).
+
+Before running Poseidon all the above three components must be running.
+
+**Note:** 
+
+   Heapster sink for poseidon is not yet merged in the heapster repo.
+   We will be doing that shortly. Please refer to the deployment scripts already created for [heapster](https://raw.githubusercontent.com/kubernetes-sigs/poseidon/master/deploy/heapster-poseidon.yaml). 
+   
+   For more info on the Heapster sink for poseidon please refer [here](https://github.com/camsas/heapster).
+   
+   
+# System requirements
+  * Go 1.9+
+  * Ubuntu 16.04
+  * Kubernetes v1.5+
+  * Docker 1.7+
+
+# Build
+
+  * **Building firmament:**
+  
+       For completeness we have included the firmament build steps here.
+     
+     
+```
+$ git clone -b dev https://github.com/Huawei-PaaS/firmament
+$ mkdir build
+$ cd build
+$ cmake ..
+$ make
+```
+
+**Note:**
+Currently the firmament repo refered in this documnet is our dev repo.
+We will be soon pointing it toward the main [repo](https://github.com/camsas/firmament).
+
+  * **Building Poseidon:**
+  
+  
+ ```
+ $ mkdir -p $GOPATH/src/k8s.io
+ $ cd $GOPATH/src/k8s.io
+ $ git clone https://github.com/kubernetes-sigs/poseidon
+ $ go build
+ 
+ ```
+  
+ # Docker container Build
+ 
+   * **Building Firmament docker container:**
+```
+$ git clone -b dev https://github.com/Huawei-PaaS/firmament
+$ cd firmament/contrib
+$ ./docker-build.sh
+```
+This will create a container and push it in the local registry.
+
+
+   * **Building Poseidon docker container:**
+   
+```
+$ git clone https://github.com/kubernetes-sigs/poseidon
+$ cd poseidon/deploy
+$ ./build_docker_image.sh
+```
+This will create a container and push it in the local registry.
+
+
+# Running
+  * **Running firmament as a process:**
+  
+```
+$ cd firmament
+$ ./build/src/firmament_scheduler --flagfile=config/firmament_scheduler.cfg
+
+```
+For more information on the arguments that could be passed to firmament [please refer](https://github.com/Huawei-PaaS/firmament#using-the-flow-scheduler)
+
+One can also use the below to get the list of arugments supprted by firmament.
+
+```
+./build/src/firmament_scheduler --help
+```
+
+  * **Running Poseidon as a process:**
+      
+      To run Poseidon as an independent process, it requires the config file to connect to the kubernetes api server and firmament address.

```

Comment:
it requires kubeconfig (file) and firmament's endpoint to be supplied as arguments.